src/libutils/oglapphelpers/oglapp.h

```diff
@@ -141,11 +143,61 @@ class OglApp
     unsigned int m_imageTexID;
 
     OpenGLBuilderRcPtr m_oglBuilder;
-
 };
 
 typedef OCIO_SHARED_PTR<OglApp> OglAppRcPtr;
 
+class ScreenApp: public OglApp
+{
+public:
+    ScreenApp() = delete;
+    ScreenApp(const char * winTitle, int winWidth, int winHeight);
+
+    ~ScreenApp();
+    void redisplay() override;
+    void printGLInfo() const noexcept override;
+private:
+    // Window identifier returned by glutCreateWindow.
+    int m_mainWin{ 0 };
+};
+
+#ifdef OCIO_HEADLESS_ENABLED
+
+#include <EGL/egl.h>
+
+class HeadlessApp: public OglApp
+{
+public:
+    HeadlessApp() = delete;
+    HeadlessApp(const char * winTitle, int winWidth, int winHeight);
+
+    ~HeadlessApp();
+
+    void printGLInfo() const noexcept override;
+    // Helper to print EGL info.
+    void printEGLInfo() const noexcept;
+
+    void redisplay() override;
+private:
+    EGLint m_pixBufferWidth{ 0 };
+    EGLint m_pixBufferHeight{ 0 };
+    std::vector<EGLint> m_pixBufferAttribs;
+
+    EGLDisplay m_eglDisplay;
+    EGLSurface m_eglSurface;
+    EGLConfig m_eglConfig;
+    EGLContext m_eglContext;
+
+    std::vector<EGLint> m_configAttribs;
+};
+
+#endif
+
+// Factory function for creating OglAppRcPtr
+// Returns OCIO_SHARED_PTR<HeadlessApp> if OCIO_HEADLESS_ENABLED
+OglAppRcPtr getOglAppPtr(const char * winTitle, int winWidth, int winHeight);
```

Comment:
The method should be localized with the `OglApp` class i.e. a public static method of `OglApp` class, and the name should explain what it does (and not what it returns) i.e. `createOglAp()`.

Reply:
Please do not forget to update the comment at line 29.

Reply:
This line must be removed now.